druid/src/contexts.rs

```diff
@@ -449,6 +449,21 @@ impl EventCtx<'_, '_> {
         self.widget_state.request_focus = Some(FocusChange::Focus(id));
     }
 
+    /// Transfer focus to the widget with the given `WidgetId`.
+    ///
+    /// This should only be called by a widget that currently has focus.
+    ///
+    /// See [`is_focused`] for more information about focus.
+    ///
+    /// [`is_focused`]: struct.EventCtx.html#method.is_focused
+    pub fn set_focus(&mut self, target: WidgetId) {
+        if self.is_focused() {
```

Comment:
I'm not sure this is really needed. It is relevant for "next" style navigation, but I can imagine some action that's not necessarily from a focused widget causing focus to be directed to, eg, a textbox.

Reply:
Good point.